src/observers/link_click_observer.ts

```diff
@@ -35,7 +36,7 @@ export class LinkClickObserver {
   clickBubbled = (event: MouseEvent) => {
     if (this.clickEventIsSignificant(event)) {
       const link = this.findLinkFromClickTarget(event.target)
-      if (link) {
+      if (link && elementIsNavigable(link)) {
```

Comment:
I'm not proficient enough in TypeScript to write this function in a way that returns a `boolean` _and_ verifies that the `link` argument is a non-null in a way that makes the `link &&` part of the conditional unnecessary. Is that possible?

Reply:
Not rlly, all non null checks and stuff can't be done with TS cuz it gets removed at compile time. Or I am just not understanding the question....

Reply:
The `link &&` portion of the conditional is required for the method calls within the conditional to operate with confidence that their arguments are present (unless marked as optional or `| undefined`.

I know it's possible within TypeScript to declare functions in a way that hints to the typechecker about an instances presence.

All of this might be moot, since I realize that the submitter is always optional, so the function might not be able to hint the argument's presence all of the time.

Reply:
Oh now I think I understand your question, you could try `link as Element`. That should do it